### PR TITLE
key_pair: add function to retrieve signature algorithm

### PR DESCRIPTION
**Pull request by tindzk**
## Purpose
Currently, the `alg` field is restricted to the crate's scope. 

## Changes
Provide a public function to access signature algorithm.

## Checklist
- [x] Self-review has been completed
- [x] Changes are complete and ready for review
- [x] Changes are covered by existing or new tests

### CODE REVIEW

File: src/key_pair.rs [hunk 1]
```diff
@@ -93,7 +97,7 @@ impl KeyPair {
 	/// Usually, calling this function is not neccessary and you can just call
 	/// [`from_der`](Self::from_der) instead. That function will try to figure
 	/// out a fitting [`SignatureAlgorithm`] for the given
-	/// key pair. However sometimes multiple signature algorithms fit for the
+	/// key pair. However, sometimes multiple signature algorithms fit for the
```
Comment (cpu) [inline src/key_pair.rs:100]: nit: it would be nice to have these unrelated comment fixes in a separate commit.

**Review by cpu — APPROVED** [review]
This seems reasonable to me, thanks!

File: src/key_pair.rs (hunk 1 above)
Comment (est31) [reply src/key_pair.rs:100]: the same commit is fine as long as the number of unrelated fixes aren't too large.

**Review by est31 — APPROVED** [review]

**Comment by cpu** [comment]
> ci / Documentation (nightly) (pull_request) Failing after 26s

Going to administratively merge this. The `num-traits` crate [fixed](https://github.com/rust-num/num-traits/pull/286) the `custom_code_classes_in_docs` error w/ nightly but it isn't in a released version yet.
